File: Sample Projects/Calculator/Demo/app/src/main/java/com/example/demo/MainActivity.java
package com.example.demo;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    Button btnAdd;
    EditText etFirstNumber, etSecondNumber;

    TextView tvResult;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        btnAdd = findViewById(R.id.btnAdd);
        etFirstNumber = findViewById(R.id.etFirstNumber);
        etSecondNumber = findViewById(R.id.etSecondNumber);
        tvResult = findViewById(R.id.tvResult);

        btnAdd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String firstNumber = etFirstNumber.getText().toString();
                int firstNum = Integer.parseInt(firstNumber);

                String secondNumber = etSecondNumber.getText().toString();
                int secondNum = Integer.parseInt(secondNumber);

                int temp = firstNum + secondNum;
                String result = Integer.toString(temp);
                tvResult.setText(result);

            }
        });
    }
}

File: 4 - Intent/kotlin implementation/Implicit Intent/ImplicitIntentDemo/app/src/main/java/com/example/implicitintentdemo/MainActivity.kt
package com.example.implicitintentdemo

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import androidx.activity.enableEdgeToEdge
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {

    private lateinit var pickImageLauncher: ActivityResultLauncher<Intent>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        val btnTakePhoto: Button = findViewById(R.id.btnTakePhoto)
        val ivPhoto: ImageView = findViewById(R.id.ivPhoto)

        // Register the ActivityResultLauncher
        pickImageLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
            if (result.resultCode == Activity.RESULT_OK) {
                val uri = result.data?.data
                ivPhoto.setImageURI(uri)
            }
        }

        // Launch image picker
        btnTakePhoto.setOnClickListener {
            val intent = Intent(Intent.ACTION_GET_CONTENT).apply {
                type = "image/*"
            }
            pickImageLauncher.launch(intent)
        }
    }
}
